### PR TITLE
Deployable architecture

### PR DESCRIPTION
**Pull request by shemau**
### Description

Implement standard solution in preparation for deployable architecture.

This remains an incubating release, so a patch release is appropriate.

THIS IS AN INCUBATING MODULE and this a BREAKING change. Inputs and outputs to modules and sub modules change. UPGRADE TESTING will not work and are intentionally SKIPPED.

### Release required?

- [ ] No release
- [x] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

Add standard solution

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by shemau** [comment]
THIS IS AN INCUBATING MODULE and this a BREAKING change. Inputs and outputs to modules and sub modules change. UPGRADE TESTING will not work and are intentionally SKIPPED.

File: tests/pr_test.go [hunk 1]
```diff
@@ -66,3 +68,36 @@ func TestRunUpgradeExample(t *testing.T) {
 		assert.NotNil(t, output, "Expected some output")
 	}
 }
+
```
Comment (ocofaigh) [inline tests/pr_test.go:71]: I suggest changing the upgrade test (`TestRunUpgradeExample`) to point to the DA code

File: solutions/standard/outputs.tf [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  value       = local.create_queue_manager ? var.queue_manager_name : var.existing_queue_manager_name
+  description = "Queue manager name"
+}
+
+output "queue_manager_id" {
+  value       = local.create_queue_manager ? module.queue_manager[0].id : data.ibm_mqcloud_queue_manager.queue_manager[0].id
+  description = "Queue manager ID"
+}
+
+# REVIEWER CHECK!!! Should all queue manager fields be output since this is a Deployable Architecture
```
Comment (ocofaigh) [inline solutions/standard/outputs.tf:34]: No harm in outputting as many things as possible in the DA

File: solutions/standard/catalogValidationValues.json.template [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+  "ibmcloud_api_key": $VALIDATION_APIKEY,
+  "region": "us-east",
+  "resource_group_name": $PREFIX,
+  "existing_mq_capacity_crn": $HPCS_US_SOUTH_CRN,
```
Comment (ocofaigh) [inline solutions/standard/catalogValidationValues.json.template:5]: Ah I'm going to have to add support to the catalog pipeline to pull the MQ capacity CRN here - I created an internal issue

File: solutions/standard/catalogValidationValues.json.template [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+  "deployment_name": "da-mq-instance",
```
Comment (ocofaigh) [inline solutions/standard/catalogValidationValues.json.template:6]: Does this have to be unique? If so, maybe go with `$PREFIX` as the value?

File: ibm_catalog.json [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+          {
+            "label": "Standard",
+            "name": "standard",
+            "install_type": "fullstack",
+            "working_directory": "solutions/standard",
+            "compliance": {
+              "authority": "scc-v3",
+              "profiles": [
+                {
+                  "profile_name": "CIS IBM Cloud Foundations Benchmark",
```
Comment (ocofaigh) [inline ibm_catalog.json:44]: Is there even any SCC rules in any profiles for anything the DA deploys? If there is not, there is no point in adding anything for compliance here (can be set to `"compliance": {}`).

Also this profile is deprecated. The latest one is called `CIS IBM Cloud Foundations Benchmark v1.1.0` (yes with the version in the name) and the version is `1.1.0`

**Review by ocofaigh — CHANGES_REQUESTED** [review]
Left some comments. Also your missing the update needed in `.releaserc` to auto trigger the catalog pipeline

File: .catalog-onboard-pipeline.yaml [hunk 6]
```diff
@@ -0,0 +1,14 @@
+---
+apiVersion: v1
+offerings:
+  - name: deploy-arch-ibm-mqcloud
```
Comment (ocofaigh) [inline .catalog-onboard-pipeline.yaml:4]: this needs to match the value in ibm_catalog.json which is `deploy-arch-ibm-mq-cloud`. This is the value we set up in catalog so needs to match here

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**